crates/sui-keys/src/random_names.rs

```diff
@@ -0,0 +1,219 @@
+/// This library provides two functions to generate
+/// a random combination of an adjective
+/// and a precious stone name as a well formatted
+/// string, or a list of these strings.
+
+/// A list of adjectives
+const LEFT_NAMES: [&str; 108] = [
+    "admiring",
+    "adoring",
+    "affectionate",
+    "agitated",
+    "amazing",
+    "angry",
+    "awesome",
+    "beautiful",
+    "blissful",
+    "bold",
+    "boring",
+    "brave",
+    "busy",
+    "charming",
+    "clever",
+    "compassionate",
+    "competent",
+    "condescending",
+    "confident",
+    "cool",
+    "cranky",
+    "crazy",
+    "dazzling",
+    "determined",
+    "distracted",
+    "dreamy",
+    "eager",
+    "ecstatic",
+    "elastic",
+    "elated",
+    "elegant",
+    "eloquent",
+    "epic",
+    "exciting",
+    "fervent",
+    "festive",
+    "flamboyant",
+    "focused",
+    "friendly",
+    "frosty",
+    "funny",
+    "gallant",
+    "gifted",
+    "goofy",
+    "gracious",
+    "great",
+    "happy",
+    "hardcore",
+    "heuristic",
+    "hopeful",
+    "hungry",
+    "infallible",
+    "inspiring",
+    "intelligent",
+    "interesting",
+    "jolly",
+    "jovial",
+    "keen",
+    "kind",
+    "laughing",
+    "loving",
+    "lucid",
+    "magical",
+    "modest",
+    "musing",
+    "mystifying",
+    "naughty",
+    "nervous",
+    "nice",
+    "nifty",
+    "nostalgic",
+    "objective",
+    "optimistic",
+    "peaceful",
+    "pedantic",
+    "pensive",
+    "practical",
+    "priceless",
+    "quirky",
+    "quizzical",
+    "recursing",
+    "relaxed",
+    "reverent",
+    "romantic",
+    "sad",
+    "serene",
+    "sharp",
+    "silly",
+    "sleepy",
+    "stoic",
+    "strange",
+    "stupefied",
+    "suspicious",
+    "sweet",
+    "tender",
+    "thirsty",
+    "trusting",
+    "unruffled",
+    "upbeat",
+    "vibrant",
+    "vigilant",
+    "vigorous",
+    "wizardly",
+    "wonderful",
+    "xenodochial",
+    "youthful",
+    "zealous",
+    "zen",
+];
+
+const LEFT_LENGTH: usize = LEFT_NAMES.len();
+
+/// A list of precious stones
+const RIGHT_NAMES: [&str; 53] = [
+    "agates",
+    "alexandrite",
+    "amber",
+    "amethyst",
+    "apatite",
+    "avanturine",
+    "axinite",
+    "beryl",
+    "beryl",
+    "carnelian",
+    "chalcedony",
+    "chrysoberyl",
+    "chrysolite",
+    "chrysoprase",
+    "coral",
+    "corundum",
+    "crocidolite",
+    "cyanite",
+    "cymophane",
+    "diamond",
+    "dichroite",
+    "emerald",
+    "epidote",
+    "euclase",
+    "felspar",
+    "garnet",
+    "heliotrope",
+    "hematite",
+    "hiddenite",
+    "hypersthene",
+    "idocrase",
+    "jasper",
+    "jet",
+    "labradorite",
+    "malachite",
+    "moonstone",
+    "obsidian",
+    "opal",
+    "pearl",
+    "phenacite",
+    "plasma",
+    "prase",
+    "quartz",
+    "ruby",
+    "sapphire",
+    "sphene",
+    "spinel",
+    "spodumene",
+    "sunstone",
+    "topaz",
+    "tourmaline",
+    "turquois",
+    "zircon",
```

Comment:
ooc where does these words come from? 

maybe i'm not familiar with the context, but when i see aliasing i would imagine as a user i can give alias to the keys (ie test_1) instead of a randomly generated one?

Reply:
Both will be supported, setting an alias or randomly being assigned an alias. One of the decisions I took was to have an alias for each key. Not sure if this is the right decision or not (and maybe it's not, we shouldn't force the user to always have an alias... if the user wants to remove the alias, it should be allowed to do so)
To directly answer your question: these are simply gemstones names. It is similar to what `docker` does with their containers that get assigned a random name if one is not given.

In a follow up PR I will push a command `sui keytool alias` that will allow to add, delete, retrieve, or rename an alias.